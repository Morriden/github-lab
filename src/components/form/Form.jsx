import React from 'react';
import PropTypes from 'prop-types';

const Form = ({ userName, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value={userName} onChange={onChange} />
    <button>Submit</button>
  </form>
);

Form.propTypes = {
  userName: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Form;
