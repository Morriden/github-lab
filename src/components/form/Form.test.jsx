import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Form userName="Morriden" onChange={() => {}} onSubmit={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
