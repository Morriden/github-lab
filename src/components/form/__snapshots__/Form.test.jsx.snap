// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders App 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="Morriden"
  />
  <button>
    Submit
  </button>
</form>
`;

exports[`ShortForm component renders App 1`] = `
<form>
  <input
    onChange={Object {}}
    type="text"
    value={
      Object {
        "onChange": [Function],
        "onSubmit": [Function],
        "userName": "Morriden",
      }
    }
  />
  <button>
    Submit
  </button>
</form>
`;
